test-apps/shared/src/main/java/com/google/gcloud/tests/ServiceTests.java

```diff
@@ -0,0 +1,178 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.tests;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.gcloud.AuthCredentials;
+import com.google.gcloud.bigquery.BaseTableInfo;
+import com.google.gcloud.bigquery.BigQuery;
+import com.google.gcloud.bigquery.BigQueryOptions;
+import com.google.gcloud.bigquery.DatasetInfo;
+import com.google.gcloud.bigquery.Field;
+import com.google.gcloud.bigquery.Schema;
+import com.google.gcloud.bigquery.TableId;
+import com.google.gcloud.bigquery.TableInfo;
+import com.google.gcloud.bigquery.testing.RemoteBigQueryHelper;
+import com.google.gcloud.datastore.Datastore;
+import com.google.gcloud.datastore.DatastoreOptions;
+import com.google.gcloud.datastore.DateTime;
+import com.google.gcloud.datastore.Entity;
+import com.google.gcloud.datastore.Key;
+import com.google.gcloud.datastore.KeyFactory;
+import com.google.gcloud.resourcemanager.ProjectInfo;
+import com.google.gcloud.resourcemanager.ResourceManager;
+import com.google.gcloud.resourcemanager.ResourceManagerOptions;
+import com.google.gcloud.storage.Blob;
+import com.google.gcloud.storage.BlobId;
+import com.google.gcloud.storage.BlobInfo;
+import com.google.gcloud.storage.BucketInfo;
+import com.google.gcloud.storage.Storage;
+import com.google.gcloud.storage.StorageOptions;
+import com.google.gcloud.storage.testing.RemoteGcsHelper;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.ByteBuffer;
+import java.nio.channels.WritableByteChannel;
+import java.util.Iterator;
+import java.util.Set;
+
+public class ServiceTests {
+
+  public static final Set<String> SUPPORTED_SERVICES =
+      ImmutableSet.of("bigquery", "datastore", "resourcemanager", "storage");
+
+  private static void runBigQueryActions(BigQuery bigquery, PrintWriter pw) {
+    pw.println("Testing BigQuery.");
+    String datasetName = RemoteBigQueryHelper.generateDatasetName();
+    bigquery.create(DatasetInfo.builder(datasetName).build());
+    TableId tableId = TableId.of(datasetName, "my_table");
+    BaseTableInfo info = bigquery.getTable(tableId);
+    if (info == null) {
+      pw.println("Creating table " + tableId);
+      Field integerField = Field.of("fieldName", Field.Type.integer());
+      bigquery.create(TableInfo.of(tableId, Schema.of(integerField)));
+    }
+    RemoteBigQueryHelper.forceDelete(bigquery, datasetName);
```

Comment:
A comment, saying that this test will create "my_table" and **delete** it at the end is important.
Also, maybe we should find an even more obscured name.